adafruit_us100.py

```diff
@@ -51,34 +53,52 @@ def __init__(self, uart):
     def distance(self):
         """Return the distance measured by the sensor in cm.
         This is the function that will be called most often in user code.
-        If no signal is received, we'll throw a RuntimeError exception. This means
-        either the sensor was moving too fast to be pointing in the right
+        If no signal is received, return ``None``. This can happen when the
+        object in front of the sensor is too close, the wiring is incorrect or the
+        sensor is not found. If the signal received is not 2 bytes, return ``None``.
+        This means either the sensor was moving too fast to be pointing in the right
         direction to pick up the ultrasonic signal when it bounced back (less
         likely), or the object off of which the signal bounced is too far away
-        for the sensor to handle. In my experience, the sensor can detect
+        for the sensor to handle. In my experience, the sensor can not detect
         objects over 460 cm away.
         :return: Distance in centimeters.
         :rtype: float
         """
-        self._uart.write(bytes([0x55]))
-        data = self._uart.read(2)  # 2 bytes return for distance
-        if not data:
-            raise RuntimeError("Sensor not found. Check your wiring!")
+        for _ in range(2):  # Attempt to read twice.
+            self._uart.write(bytes([0x55]))
+            data = self._uart.read(2)  # 2 byte return for distance.
```

Comment:
add a delay here, maybe time.sleep(0.01) or time.sleep(0.1) - it takes time to get readings from sonar

Reply:
The `_uart.read()` will wait for whatever UART read timeout is specified when it was created (default 1 second). Then, if we still don't get a reading, the code sleeps for 100ms on line 72 and tries again. https://github.com/stoduk/PingSerial talks about how asking the sensor too soon after a previous reading causes it to ignore the request. Would you like a 100ms delay up front to avoid depending on the UART timeout? We could add `time.sleep(0.1) between lines 68 and 69.

Reply:
oh right - i was hoping to fix the flaky behavior...i guess its just flaky

Reply:
@kattni well i think it's still a good idea to add a `time.sleep(0.1)` after each `self._uart.write()`, because otherwise we're depending on the UART timeout, which can be changed by the user (it might be too short). And we'll still keep the other `time.sleep()`.